AirCasting/CoreData/NSManagedObjectContext+Utils.swift

```diff
@@ -40,10 +40,17 @@ extension NSManagedObjectContext {
     }
     
     // Checks if stream exists, if not, creates a new one
-    func newOrExisting<T: NSManagedObject>(streamID: MeasurementStreamID) throws -> T  {
+    #warning("This seems odd and is also working odd. I don't think this is correct implementation")
```

Comment:
Do you have an idea for a better implementation? Cause I wouldn't know how to proceed with this warning and we are supposed to decrease the number of warnings since we already have a lot. 😅

Reply:
The warning is there to be fixed asap, but this is too big of an issue to do it along this PR. I think we need to establish some sort of a scheme for dealing with those. I'd suggest:
**For potential bugs or severe code quality issues**
1) Place warnings in places that are *bad bad* (potentially breaking stuff, like here) and too big to resolve ad hoc.
2) When placing warning, add a comprehensive explanation of an issue in the comment. Also link the ticket from 3) in this comment.
3) Add a ticket to github issues section with correct tag (I suggest the tag "warning" + another one indicating severity high/mid/low or smth)

**For less severe stuff like minor code quality issues**
1) When the issue is not *bad bad*, but to big to resolve ad hoc, add a `// FIXME: ` marking, but explain it really well so that someone with more time that stumbles across this will be able to refactor/fix. Also link the ticket from 2) in this comment.
2) Add a ticket to github issues section with correct tag (I suggest the tag "quality" for quality issues and other tags for other issues. We can also create tags for parts of the app or to indicate if resolving this will require AirBeam).

What do you think about this @anna1901 @jankrzempek ? If/when we decide on some kind of a more formalized approach we shall add a section to the `README.md` file.

Reply:
I love the idea! This will help a lot to control all the warnings and issues that pop up and to start gathering parts of the app that need refactoring in one place.

Reply:
Could you post this proposition on slack as well? Maybe we could move the discussion there and then pin it for future reference.